python/pyspark/pandas/data_type_ops/num_ops.py

```diff
@@ -247,14 +247,22 @@ def truediv(self, left: IndexOpsLike, right: Any) -> SeriesOrIndex:
         _sanitize_list_like(right)
         if not is_valid_operand_for_numeric_arithmetic(right):
             raise TypeError("True division can not be applied to given types.")
+        right = transform_boolean_operand_to_numeric(right, spark_type=left.spark.data_type)
 
         def truediv(left: PySparkColumn, right: Any) -> PySparkColumn:
-            return F.when(
-                F.lit(right != 0) | F.lit(right).isNull(),
-                left.__div__(right),
-            ).otherwise(F.lit(np.inf).__div__(left))
+            if not get_option("compute.ansi_mode_support"):
```

Comment:
can we optimize out this `get_option` which needs a separate Config RPC?
I guess we can just use the new branch